### PR TITLE
Civ 2700 dq support

### PR DESCRIPTION
**Pull request by osheaf01**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-2700

### Change description ###
Direct Questionnaire - support you'd require for a court hearing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/common/models/claim.ts [hunk 1]
```diff
@@ -135,38 +137,49 @@ export class Claim {
...
   }
+
   isChildrenDisabled(): boolean {
     return this.statementOfMeans?.dependants?.declared === true &&
       this.statementOfMeans?.childrenDisability?.option === YesNo.YES;
   }
+
   isDefendantSeverelyDisabledOrDependentsDisabled(): boolean {
-    return this.isChildrenDisabled() || this.isPartnerDisabled() || this.isDefendantDisabledAndSeverlyDiabled();
+    return this.isChildrenDisabled() || this.isPartnerDisabled() || this.isDefendantDisabledAndSeverelyDiabled();
```
Comment (TatsyMaslova) [inline src/main/common/models/claim.ts:164]: typo in .isDefendantDisabledAndSeverelyDiabled() should be disabled 😄

Comment (osheaf01) [reply src/main/common/models/claim.ts:164]: Ha, I catch one typo and miss another. Will fix!

File: src/main/services/features/directionsQuestionnaire/supportRequiredService.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('supportRequiredService');
+
+
+export const getSupportRequired = async (claimId: string): Promise<SupportRequired> => {
+  try {
+    let supportRequired;
+    const case_data = await getCaseDataFromStore(claimId);
+    if (!case_data?.supportRequired) {
```
Comment (GorilaNaranja) [inline src/main/services/features/directionsQuestionnaire/supportRequiredService.ts:12]: I think we don't need ? in case_data, getCaseDataFromStore always return a Claim

`if (!case_data.supportRequired)`

File: src/main/services/features/directionsQuestionnaire/supportRequiredService.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
+};
+
+export const saveSupportRequired = async (claimId: string, supportRequired: SupportRequired) => {
+  try {
+    const case_data = await getCaseDataFromStore(claimId);
+    if (!case_data?.supportRequired) {
```
Comment (GorilaNaranja) [inline src/main/services/features/directionsQuestionnaire/supportRequiredService.ts:27]: Same here

File: src/main/views/features/directionsQuestionnaire/support-required.njk [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+{% block pageTitle %}
+  {{ setPageTitle('PAGES.SUPPORT_REQUIRED.PAGE_TITLE') }}
+{% endblock %}
+
+{% block content %}
+  {% set errors = form.getErrors() %}
+  {% if form.hasErrors() %}
+    {{ govukErrorSummary({
+      "titleText": t("There was a problem"),
```
Comment (GorilaNaranja) [inline src/main/views/features/directionsQuestionnaire/support-required.njk:19]: Use translation key
`t("ERRORS.THERE_WAS_A_PROBLEM")`

**Review by laperla11 — APPROVED** [review]

File: src/main/views/features/directionsQuestionnaire/support-required.njk [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                  conditional: {
+                  html: checkedOtherHtml
+                },
+                  checked: true if form.model.otherSupportSelected
+                }
+              ]
+            }) }}
+            <div class="govuk-form-group">
+              {{ govukButton({
+                text: t("Save and continue")
```
Comment (GorilaNaranja) [inline src/main/views/features/directionsQuestionnaire/support-required.njk:131]: I know that is not the scope of your issue but maybe you can create a new key for "Save and continue" in COMMONS

File: src/main/routes/features/directionsQuestionnaire/supportRequiredController.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    res.status(500).send({error: error.message});
+  }
+});
+
+supportRequiredController.post(SUPPORT_REQUIRED_URL, async (req, res) => {
+  try {
+    const claimId = req.params.id;
+    let supportRequired = new SupportRequired();
+    if (req.body.declared) {
+      const languageSelected = boolean(req.body.declared.includes('languageSelected'));
```
Comment (commitBlob) [inline src/main/routes/features/directionsQuestionnaire/supportRequiredController.ts:30]: you dont need import for boolean evaluation `!!(req.body.declared.includes('signLanguageSelected'))` will do the trick

File: src/main/views/features/directionsQuestionnaire/support-required.njk (hunk 5 above)
Comment (osheaf01) [reply src/main/views/features/directionsQuestionnaire/support-required.njk:131]: Will do.

**Review by laarmada — APPROVED** [review]

**Pull request merged**